warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/EdgeEventFieldUtil.java

```diff
@@ -0,0 +1,241 @@
+        } else {
+            // If fields does not include the load date then use the current system time as load date
+            loadDateStr = DateHelper.format(new Date(now.get()));
+        }
+        return loadDateStr;
+    }
+
+    private String getEdgeAttr3(EdgeDefinitionConfigurationHelper edgeDefConfigs) {
+        // get the edgeAttribute3 from the event fields map
+        if (normalizedFields.containsKey(edgeDefConfigs.getEdgeAttribute3())) {
+            return normalizedFields.get(edgeDefConfigs.getEdgeAttribute3()).iterator().next().getIndexedFieldValue();
+        }
+        return null;
+    }
+
+    private String getEdgeAttr2(EdgeDefinitionConfigurationHelper edgeDefConfigs) {
+        // get the edgeAttribute2 from the event fields map
+        if (normalizedFields.containsKey(edgeDefConfigs.getEdgeAttribute2())) {
+            return normalizedFields.get(edgeDefConfigs.getEdgeAttribute2()).iterator().next().getIndexedFieldValue();
+        }
+        return null;
+    }
+
+    // this just avoids ugly copy pasta
+    private NormalizedContentInterface getNullKeyedNCI(String fieldValue, Multimap<String,NormalizedContentInterface> fields) {
+        Iterator<NormalizedContentInterface> nciIter = fields.get(fieldValue).iterator();
+        if (nciIter.hasNext()) {
+            return nciIter.next();
+        }
+        return null;
+    }
+
+    public void setEdgeDuration(EdgeDefinition edgeDef, EdgeDataBundle edgeDataBundle) {
+        if (edgeDef.getUDDuration()) {
+            NormalizedContentInterface upnci = getNullKeyedNCI(edgeDef.getUpTime(), normalizedFields);
+            NormalizedContentInterface downnci = getNullKeyedNCI(edgeDef.getDownTime(), normalizedFields);
+            if (null != upnci && null != downnci) {
+                edgeDataBundle.initDuration(upnci, downnci);
+            }
+        } else {
+            NormalizedContentInterface elnci = getNullKeyedNCI(edgeDef.getElapsedTime(), normalizedFields);
+            if (null != elnci) {
+                edgeDataBundle.initDuration(elnci);
+            }
+        }
+    }
+
+    protected void setEdgeKeyDateType(EdgeDataBundle bundle, boolean validActivtyDate, boolean sameActivityDate, long eventDate, long newFormatStartDate) {
+        EdgeKey.DATE_TYPE dateType = null;
+        if (eventDate < newFormatStartDate) {
+            dateType = EdgeKey.DATE_TYPE.OLD_EVENT;
+        } else if (validActivtyDate) {
+            if (sameActivityDate) {
+                dateType = EdgeKey.DATE_TYPE.ACTIVITY_AND_EVENT;
+            } else {
+                dateType = EdgeKey.DATE_TYPE.ACTIVITY_ONLY;
+                // also need to write EVENT_ONLY for some reason
+            }
+        } else {
+            dateType = EdgeKey.DATE_TYPE.EVENT_ONLY;
+        }
+        bundle.setDateType(dateType);
+    }
+
+    public EdgeDataBundle setEdgeInfoFromEventFields(EdgeDataBundle bundle, EdgeDefinitionConfigurationHelper edgeDefConfigs, RawRecordContainer event,
```

Comment:
This method returns the bundle it received by reference.  The return value itself is never used.  Perhaps the method should just create a new bundle from the provided event and drop the first parameter.